### ISSUE TITLE
Tabbed browser sync 

### ISSUE THREAD

**Issue by bijang** [xbrowsersync/app#128]
For the planned tab sync feature, as well as, or instead of, sync in the regular fashion, would it be possible to sync data coming out of tabsessionmanager (https://github.com/sienori/Tab-Session-Manager), which is essentially a continuum, set to your own specification, of snapshots of your tabbed browsing session; so essentially you can have a 'journal log' of your activity to roll back to. This would save developing this infrastructure on your side by leveraging this project, which I personally use mainly becasue it can leverage another great project, treestyletab (https://github.com/piroor/treestyletab) , which adds tremendous value to your tab history/session browse. If the dependency on another app is unattractive, could I ask it be made an option at least?

**Comment by nero120** [comment]
Thanks for the suggestion @bijang, looks interesting. Ince I'm ready to implement syncing tabs I will consider this carefully!

**Comment by r4taw2mg** [comment]
Hi. After a short research on synching open tabs across browsers/devices, I have landed here. Currently it seems, like there is no app or service available offering this service, right? Only synching across identical browser engines (Chrome, Firefox) seems to be available. I still hope, I am wrong. Any pointers will be appreciated.

**Comment by bijang** [comment]
seems like that, this project is promising but not much activity for likely very valid real world reasons,
IMHO i think it would need a monetary model, whatever that may be, like etesync or bitwarden, to first pay for a 3rd party audit of their security, and then maybe other developers to implement tab/otherfeature support, 
I don't yet use etesync because of a (planned) audit; similarly bitwarden recieved a lot of criticism for  same until they recently did one, now its the hot favourite. 
This project as far as i can see is not sustainable without fundamental funding changes, which is a shame because it's a great potential

**Comment by r4taw2mg** [comment]
I, too, see the possibilities. I am not sure, a security audit has to be one of the first steps, but developer engagement is always critical. So many promising FOSS projects start with a lot of motivation but dying of sooner or later. So many, I started giving donations to stuff, I use regularly. But finding ways of monetization while staying true to 'free' is hard.

**Comment by bijang** [comment]
yes its hard to ask for money, but not impossible or unethical and the afformented examples are a testament and example to that; though security audit, first step or not, is mandatory, it wouldnt be acceptable as a serious "secure" project, from a PR point of view at minimum, more imporatantly from an objective security point of view,.... I think as mature users we have to evaluate the benefites of paying a small fee for a good alternative to privacy "free" projects where privacy is compromised/sold

**Comment by nero120** [comment]
I appreciate the concerns, indeed it is the nature of FOSS that devs contribute to them in their spare time as they are by definition not businesses with a profit model to incentivise investment of time. In my own case I would love to spend more time on xBrowserSync and finish implementing the requested features, I fully intend to but currently have limited time to spend on it, hence why the current release has taken longer than hoped.

Rest assured I'm still committed to the project and fully intend to continue working on it as I'm able.

**Comment by r4taw2mg** [comment]
Please don't get me wrong. I appreciate the work you are doing and I wish you lots of fun and lots of success and that your professional and private environment lets you concentrate on the things that you like to do. I was just referring to the changes that affect our lifes. Sometimes this make shifting attention and energy necessary. And sometimes people just move on.

**Comment by bijang** [comment]
Agree the project and work is very appreciated

**Comment by nero120** [comment]
Thanks guys 🙇

**Comment by Leopere** [comment]
I'm personally heavily reliant on tab syncing and I really want something that doesn't appear to be destined to be premium like Workona.  This feature set has been coming and going forever google had browser-sync like this 10+years ago.

**Comment by leopq** [comment]
Hi, there! Since I've started to work remotely due to COVID-19, I've started to work on different computers around the house during the work day, so keeping things synced became a "thing" for me. Yesterday, I've heard about the xBrowserSync and installed the extension on Chrome, Edge and Opera (On MacOS, Linux and Windows), so:

- So far, I'm simply amazed by the concerns with security and the *beautifully* crafted retro design.
- When I've read that this is an open source project, my impression grew even more.
- When I've realized that it was built with a cross platform stack based on JS, I'm simply in love with it ❤️

I have a very restricted time schedule right now due to my work demands and college graduation project, but I'll put some effort into researching this tab sync feature. In case I find anything relevant or make some proof of concept over this, I'll make sure to keep you guys posted!

PS.: Lots of good points towards FOSS and its future - which is a necessary and widely relevant debate, by the way.

**Comment by nero120** [comment]
@leopq thank you for the kind comments! v1.6.0 is going to drop soon which features a completely refactored codebase using typescript and a better code structure, this will also mean other devs will find it much easier to contribute code to the project.

Any effort you can give to the project is gratefully accepted and appreciated. Ultimately if xBrowserSync cements its place as **the** open source browser syncing tool I would hope many devs are contributing to the maintenance and improvement of the project, it cannot rely on just one dev. However, for the time being it is my responsibility to get it to [v1 feature complete](https://github.com/xbrowsersync/app/wiki/Roadmap) at the very least. If you can help get it to that point quicker that would be fantastic!!

**Comment by DOFfactory** [comment]
Please keep us posted about the updates on this feature. Much needed!
